src/App.jsx

```diff
@@ -7,19 +7,35 @@ import database from './database.json';
 import './App.css';
 
 function App() {
+  const [productsList, setProductsList] = useState(database);
+
+  const upvoteProduct = (id) => {
+    const productListClone = [...productsList];
```

Comment:
Given that `productList` is a piece of ReactJS state, we cant perform _in-place_/_in-situ_ operations on it, we must first clone it and work on the cloned data.